lib/index.js

```diff
@@ -306,7 +311,11 @@ exports.validate = internals.validate = function (email, options, callback) {
     let token;                                      // Token is used outside the loop, must declare similarly
     for (let i = 0; i < emailLength; i += token.length) {
         // Utilize codepoints to account for Unicode surrogate pairs
-        token = String.fromCodePoint(email.codePointAt(i));
+        const codePoint = email.codePointAt(i);
+        if (!allowUnicode && codePoint > 0x7f) {
+            return false;
```

Comment:
This ignores the `errorLevel` option - at the very least, it should output an appropriate code when `diagnose` is `true`.

This demonstrates the imperfections of the errorLevel system as implemented, because the user might set `allowUnicode` to `true`, but set the `errorLevel` higher than the error code corresponding to disallowed unicode.

I guess we could also just note that this is the behavior for `allowUnicode`, and that `errorLevel` isn't really compatible with it.

Reply:
I tried to map it to an `errorLevel` value here: https://github.com/hapijs/isemail/pull/160/commits/c3430a0f3103040b2691d6bee2efc90603bdaf41

Let me know what you think.